### PR TITLE
add warning for incorrect casing on autofocus #3248

### PR DESCRIPTION
**Pull request by jonhester**
Includes test.

This is my first pull request to react. I have completed the CLA.

### CODE REVIEW

**Comment by jonhester** [comment]
Travis is failing on lint task, none of the errors shown are related to this event. Running grunt lint on current master also fails. See https://github.com/facebook/react/issues/3270

File: src/browser/ui/dom/DOMProperty.js [hunk 1]
```diff
@@ -141,6 +142,19 @@ var DOMPropertyInjection = {
         propName
       );
     }
+
+    for (var polyfillName in Polyfills) {
+      DOMProperty.isPolyfillName[polyfillName] = true;
+
+      if (DOMAttributeNames.hasOwnProperty(polyfillName)) {
+        var attributeName = DOMAttributeNames[polyfillName];
```
Comment (Pouja) [inline src/browser/ui/dom/DOMProperty.js]: attributeName is already defined. See line 269 in the Travis CI test of your PR.

Comment (jonhester) [inline src/browser/ui/dom/DOMProperty.js]: Ahhh...thanks. I'll fix.

**Comment by jonhester** [comment]
I fixed the linting issues last week. Any thoughts on this?

**Comment by zpao** [comment]
At a glance this seems ok, though it's not great that we're adding code that will ship to prod which is only useful for development warnings. So we might want to try something else. We're code frozen for 0.13 so it won't go in there (though perhaps patch release). We'll take a closer look at this and a few other issues after we cut the release branch.

**Pull request closed without merging**